### PR TITLE
Button styling refactor & regression tests

### PR DESCRIPTION
**Pull request by Stanton**
This begun as a ticket to resolve an issue with outline button label colours, and uncovered a whole host of issues with various combinations of styles so ended up refactoring a big chunk of the button scss. 

The following main colour variations are:

* `.btn`
* `.btn--primary`
* `.btn--success`
* `.btn--warning`
* `.btn--danger`
* `.btn--info`
* `.btn--inverse`
* `.btn--white` 👈🏿 NEW!

And a button can any, or a combination of the following modifiers:

* `.btn--small`
* `.btn--outline`
* `.btn--naked`

and each button has various states:

* `:active` `.active`
* `:hover` `.is-hovered`
* `:disabled` `.is-disabled`

Because of this there's a **lot** of css being spat out, I've spent some time setting up visual regression tests try try cover all of the various combinations available so if any refactoring is suggested, we can ensure no issues creep in.

# Regular buttons

screen shot 2016-10-16 at 21 00 23

# Disabled regular buttons

screen shot 2016-10-16 at 21 00 51

# Outline buttons

screen shot 2016-10-16 at 21 00 37

# Disabled outline buttons

screen shot 2016-10-16 at 21 01 01

# Small regular buttons

screen shot 2016-10-16 at 21 01 10

# Small outline buttons

screen shot 2016-10-16 at 21 01 21

### Notes for reviewer

I added a white background to all wraith screenshots, so the entire baseline has been regenerated in this PR. Sorry.

Addresses #366 #367 
Depends on #363

### CODE REVIEW

File: views/lexicon/html.html.twig [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+{%
+    set tabs_content = [
+        {
+          "id"    : "buttons",
+          "label" : "Buttons",
+          "src"   : tab_buttons,
+          "active": true
+        },
+
```
Comment (langy) [inline views/lexicon/html.html.twig]: Remove empty line and preceding trailing comma

File: stylesheets/_component.buttons.scss [hunk 2]
```diff
@@ -40,222 +48,251 @@
...
-    @include button-states('info');
 
-    .badge {
-        background-color: $badge-info-bg-darker;
-        text-shadow: 0 -1px 0 rgba(0, 0, 0, .25);
+    &:active,
+    &.active {
+        background-color: color(base, light) !important;
+        box-shadow: inset 0 0 0 2px color(base);
+        color: pick_best_color(color(base, light), (#fff, color(gray, dark))) !important;
```
Comment (langy) [inline stylesheets/_component.buttons.scss]: Are there too many parentheses here? Highlighting looks odd in the diff where the `#fff, color(gray, dark)` is wrapped, so just making sure it's intentional.

Comment (Stanton) [inline stylesheets/_component.buttons.scss]: It's intentional, yes

File: stylesheets/_component.buttons.scss [hunk 3]
```diff
@@ -40,222 +48,251 @@
...
-.btn--naked:active,
-.btn--naked > .btn,
-.btn--naked > .btn:active {
+.btn--naked > .btn {
     background: none;
     border: 0;
     box-shadow: none;
-    color: color(text);
+    color: color(text) !important;
     display: inline-block;
```
Comment (Stanton) [inline stylesheets/_component.buttons.scss]: Removed, no wraith failures 👍

File: stylesheets/_component.buttons.scss [hunk 4]
```diff
@@ -40,222 +48,251 @@
...
     line-height: auto;
```
Comment (Stanton) [inline stylesheets/_component.buttons.scss]: Removed, no wraith failures 👍

**Comment by Stanton** [comment]
@langy Suggested changes implemented

**Pull request merged**